python/ray/data/dataset.py

```diff
@@ -4059,13 +4059,15 @@ def to_tf(
         feature_columns: Union[str, List[str]],
         label_columns: Union[str, List[str]],
         *,
+        additional_columns: Union[str, List[str]] = None,
```

Comment:
I'm not thrilled about the naming of this argument. Open for brainstorming 😄 

Even though the current use case for this argument is to include sample weights, I don't think naming it as `sample_weight` or `weight` is a good idea as there may be other metadata that we want incorporate/yield in the future.

Reply:
I also deliberately allow `List[str]` as its type as we may want to associate more than one metadata to each (feature, label) pair.